client/alloc_runner.go

```diff
@@ -202,12 +202,11 @@ func (r *AllocRunner) saveAllocRunnerState() error {
 }
 
 func (r *AllocRunner) saveTaskRunnerState(tr *TaskRunner) error {
-	var err error
-	if err = tr.SaveState(); err != nil {
-		r.logger.Printf("[ERR] client: failed to save state for alloc %s task '%s': %v",
+	if err := tr.SaveState(); err != nil {
+		return fmt.Errorf("client: failed to save state for alloc %s task '%s': %v",
```

Comment:
@nak3  we use the format `subsytem: message` only when we are printing the log message. When we create an error, we simply use the error message so in this case the statement would be `fmt.Errorf("failed to save state for alloc %s task '%s': %v",r.alloc.ID, tr.task.Name, err)

I am going to merge this commit and fix this myself.